File: api-schema/org/hiring-settings.ts
export interface OrgHiringSettings {
	cool_off_days: number;
	allow_unsolicited_endorsements_default: boolean;
}

export interface UpdateOrgHiringSettingsRequest {
	cool_off_days: number;
	allow_unsolicited_endorsements_default?: boolean;
}

export function validateUpdateOrgHiringSettingsRequest(
	req: unknown
): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;

	if (typeof r.cool_off_days !== "number") {
		errors.push({ field: "cool_off_days", message: "Must be a number" });
	} else if (r.cool_off_days < 0 || r.cool_off_days > 365) {
		errors.push({
			field: "cool_off_days",
			message: "Must be between 0 and 365",
		});
	}

	if (
		r.allow_unsolicited_endorsements_default !== undefined &&
		typeof r.allow_unsolicited_endorsements_default !== "boolean"
	) {
		errors.push({
			field: "allow_unsolicited_endorsements_default",
			message: "Must be a boolean",
		});
	}

	return errors;
}

export interface ValidationError {
	field: string;
	message: string;
}

File: api-schema/org/hiring-settings.go
package org

import (
	"vetchium-api-server.typespec/common"
)

type OrgHiringSettings struct {
	CoolOffDays                         int32 `json:"cool_off_days"`
	AllowUnsolicitedEndorsementsDefault bool  `json:"allow_unsolicited_endorsements_default"`
}

type UpdateOrgHiringSettingsRequest struct {
	CoolOffDays                         int32 `json:"cool_off_days"`
	AllowUnsolicitedEndorsementsDefault *bool `json:"allow_unsolicited_endorsements_default,omitempty"`
}

// Validation functions
func (r *UpdateOrgHiringSettingsRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError

	if r.CoolOffDays < 0 || r.CoolOffDays > 365 {
		errs = append(errs, common.ValidationError{
			Field:   "cool_off_days",
			Message: "must be between 0 and 365",
		})
	}

	return errs
}
